sdk/src/cameras/3d-smart-camera/calibration_3d_smart.cpp

```diff
@@ -482,8 +482,8 @@ aditof::Status Calibration3D_Smart::distortionCorrection(uint16_t *frame,
             int x_dist = (int)(x_dist_adim * fx + cx);
             int y_dist = (int)(y_dist_adim * fy + cy);
 
-            if (x_dist >= 0 && x_dist < width && y_dist >= 0 &&
-                y_dist < height) {
+            if (x_dist >= 0 && (unsigned int)x_dist < width && y_dist >= 0 &&
```

Comment:
The problem is still there. You need to cast width to signed since we know the value won't be too high to overflow.